src/components/App.js

```diff
@@ -4,53 +4,61 @@ import { Container } from '@chakra-ui/react';
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 
 import { CartContextWrapper } from './CartContextWrapper/CartContextWrapper';
+import ProductListingFormContextWrapper from '../context/ProductListingFormContextWrapper';
```

Comment:
Is it necessary to add the wrapper to this file? By adding it here, wouldn't you be creating overhead for the rest of the app?

Reply:
Instead, the wrapper could be added to a page component in `src/pages`and the `ProductListingForm` component could be nested in this wrapper.

Reply:
I thought wrapper goes at the app level so that if any other components other than ProductListingForm needs access to this data in case its easier to get

Reply:
Definitely, but it doesn't seem like any other components need access to this data.

Reply:
fixed